AirCasting/MicrophoneSession/MicrophoneManager.swift

```diff
@@ -21,86 +18,110 @@ class MicrophoneManager: ObservableObject {
+                AVSampleRateKey: 44100.0,
+                AVNumberOfChannelsKey: 1,
+                AVEncoderAudioQualityKey: AVAudioQuality.min.rawValue
+            ]
+            
+            do {
+                try audioSession.setCategory(AVAudioSession.Category.playAndRecord)
+                try audioSession.setActive(true)
+                try recorder = AVAudioRecorder(url:url, settings: recordSettings)
+                
+            } catch {
+                return
             }
-        }
-        
-        self.session = session
-        createDBStream()
-        locationProvider.requestLocation()
-        
-        let url = URL(fileURLWithPath: "/dev/null", isDirectory: true)
-        
-        let recordSettings: [String: Any] = [
-            AVFormatIDKey: NSNumber(value: kAudioFormatAppleLossless),
-            AVSampleRateKey: 44100.0,
-            AVNumberOfChannelsKey: 1,
-            AVEncoderAudioQualityKey: AVAudioQuality.min.rawValue
-        ]
-        
-        do {
-            try audioSession.setCategory(AVAudioSession.Category.playAndRecord)
-            try audioSession.setActive(true)
-            try recorder = AVAudioRecorder(url:url, settings: recordSettings)
             
-        } catch {
-            return
+            recorder.isMeteringEnabled = true
+            recorder.record()
+            isRecording = true
+            
+            levelTimer = Timer.scheduledTimer(timeInterval: 1, target: self, selector: #selector(getMeasurement), userInfo: nil, repeats: true)
         }
-        
-        recorder.isMeteringEnabled = true
-        recorder.record()
-        isRecording = true
-        
-        levelTimer = Timer.scheduledTimer(timeInterval: 1, target: self, selector: #selector(getMeasurement), userInfo: nil, repeats: true)
     }
     
     @objc func getMeasurement() {
+        if !recorder.isRecording {
+            recorder.record()
+        }
         recorder.updateMeters()
-        
         let level = recorder.averagePower(forChannel: 0)
+        createMeasurement(value: level)
+    }
+    
+    func stopRecording() {
+        levelTimer.invalidate()
+        isRecording = false
+        recorder.stop()
+        recorder = nil
+        session?.status = Int16(SessionStatus.FINISHED.rawValue)
+        do {
+            try context.save()
+        } catch {
+            print("Error for saving session when stopping mic recording: \(error)")
+        }
+    }
+    
+    @objc func handleInterruption(notification: Notification) {
+        guard let userInfo = notification.userInfo,
+            let typeValue = userInfo[AVAudioSessionInterruptionTypeKey] as? UInt,
+            let type = AVAudioSession.InterruptionType(rawValue: typeValue) else {
+                return
+        }
+
+        switch type {
+        case .began:
+            levelTimer.invalidate()
+            session?.status = Int16(SessionStatus.DISCONNETCED.rawValue)
```

Comment:
Seams like ` Int16(SessionStatus.DISCONNETCED.rawValue)` is something that repeats itself. 
if we would hide the raw value type that is kept in the database private in the session object and expose our enum setter and getter it would definitely be safer, more readable etc. 

Please change `@NSManaged public var status: Int16` to : 
```
public var status: SessionStatus? {
        get { (value(forKey: "status") as? Int).flatMap(SessionStatus.init(rawValue:)) }
        set { setValue(newValue?.rawValue, forKey: "status") }
    }
```
But also keep in mind the other comment about not storing unfinished session in the database but storing it after successful recoding session.

Reply:
You already corrected it in your last PR so I can leave it as it is?